src/openexr.imageio/exroutput.cpp

```diff
@@ -799,22 +799,28 @@ struct ExrMeta {
         : oiioname(oiioname), exrname(exrname), exrtype(exrtype) {}
 };
 
+// Make our own type here because TypeDesc::TypeMatrix and friends may
+// not have been constructed yet. Initialization order fiasco, ick!
+static TypeDesc TypeMatrix (TypeDesc::FLOAT,TypeDesc::MATRIX44,0);
+static TypeDesc TypeTimeCode (TypeDesc::INT, TypeDesc::SCALAR, TypeDesc::TIMECODE, 2);
+static TypeDesc TypeKeyCode (TypeDesc::INT, TypeDesc::SCALAR, TypeDesc::KEYCODE, 7);
+
 static ExrMeta exr_meta_translation[] = {
     // Translate OIIO standard metadata names to OpenEXR standard names
-    ExrMeta ("worldtocamera", "worldToCamera", TypeDesc::TypeMatrix),
-    ExrMeta ("worldtoscreen", "worldToNDC", TypeDesc::TypeMatrix),
-    ExrMeta ("DateTime", "capDate", TypeDesc::TypeString),
-    ExrMeta ("ImageDescription", "comments", TypeDesc::TypeString),
-    ExrMeta ("description", "comments", TypeDesc::TypeString),
-    ExrMeta ("Copyright", "owner", TypeDesc::TypeString),
-    ExrMeta ("PixelAspectRatio", "pixelAspectRatio", TypeDesc::TypeFloat),
-    ExrMeta ("XResolution", "xDensity", TypeDesc::TypeFloat),
-    ExrMeta ("ExposureTime", "expTime", TypeDesc::TypeFloat),
-    ExrMeta ("FNumber", "aperture", TypeDesc::TypeFloat),
-    ExrMeta ("oiio:subimagename", "name", TypeDesc::TypeString),
-    ExrMeta ("openexr:dwaCompressionLevel", "dwaCompressionLevel", TypeDesc::TypeFloat),
-    ExrMeta ("smpte:TimeCode", "timeCode", TypeDesc::TypeTimeCode),
-    ExrMeta ("smpte:KeyCode", "keyCode", TypeDesc::TypeKeyCode),
+    ExrMeta ("worldtocamera", "worldToCamera", TypeMatrix),
+    ExrMeta ("worldtoscreen", "worldToNDC", TypeMatrix),
+    ExrMeta ("DateTime", "capDate", TypeDesc::STRING),
```

Comment:
TypeDesc::STRING is just an enum (i.e. an int) for the "base type".

TypeDesc(TypeDesc::STRING) works just fine, assembles the full TypeDesc on the fly (base type STRING, scalar, no array).

TypeDesc::TypeString is a full TypeDesc, constructed in typedesc.cpp. Or, in this case, not yet constructed.

Yes, I think in a C++11 world, TypeString (etc) will be constexpr fully defined in typedesc.h, and not globals initialized in typedesc.cpp.